### ISSUE TITLE
Panic when updating git package "reference \'refs/heads/master\' not found"

### ISSUE THREAD

**Issue by chocol4te** [nabijaczleweli/cargo-update#143]
`probe-run` installed with:

```
$ cargo install \
        --git https://github.com/knurling-rs/probe-run \
        --branch main \
        --features defmt
```

`cargo install-update -ag` then panics with:

```
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Error { code: -9, klass: 4, message: "reference \'refs/heads/master\' not found" }', /Users/ferdiamckeogh/.cargo/registry/src/github.com-1ecc6299db9ec823/cargo-update-4.1.1/src/ops/mod.rs:436:31
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

**Comment by nabijaczleweli** [comment]
Can you try with the current `master` branch (at least 52256cc768ecdff4376cdf3b84a846af374f0772)? That fixed it for me.

**Comment by chocol4te** [comment]
All working, thank you so much!:)

**Comment by nabijaczleweli** [comment]
Delightful! Released as [`v4.1.2`](https://github.com/nabijaczleweli/cargo-update/releases/tag/v4.1.2).
